micrometer-test/src/main/java/io/micrometer/core/tck/MeterRegistryCompatibilityKit.java

```diff
@@ -300,6 +301,68 @@ void histogramCounts() {
             assertThat(s.histogramCountAtValue(2)).isNaN();
         }
 
+        @Issue("#3904")
+        @Test
+        void histogramCountsPublishPercentileHistogramAndSlos() {
+            DistributionSummary summary = DistributionSummary.builder("my.summmary")
+                .serviceLevelObjectives(5, 50, 95)
+                .publishPercentileHistogram()
+                .register(registry);
+
+            // ensure time-window based histograms are not fully rotated when we assert
+            Duration halfStep = step().dividedBy(2);
+            clock(registry).add(halfStep);
+
+            for (int val : new int[] { 22, 55, 66, 98 }) {
+                summary.record(val);
+            }
+            // accommodate StepBucketHistogram
+            clock(registry).add(halfStep);
+
+            HistogramSnapshot snapshot = summary.takeSnapshot();
+            CountAtBucket[] countAtBuckets = snapshot.histogramCounts();
+
+            assertThat(countAtBuckets).satisfiesAnyOf(
+                    MeterRegistryCompatibilityKit.this::assertHistogramNotSupportingPercentileHistogramBuckets,
+                    MeterRegistryCompatibilityKit.this::assertHistogramSupportingPercentileHistogramBuckets);
+        }
+
+    }
+
+    private void assertHistogramNotSupportingPercentileHistogramBuckets(CountAtBucket[] countAtBuckets) {
+        // only SLO buckets added because percentile histogram buckets are not supported
+        assertThat(countAtBuckets).extracting(CountAtBucket::bucket).containsExactly(5.0, 50.0, 95.0);
+
+        assertThat(countAtBuckets).extracting(CountAtBucket::count).containsExactly(0.0, 1.0, 3.0);
```

Comment:
There's unfortunately no point in the assertions in this method since the other method will always be true when these assertions are true, and we need the lax conditions there for the reasons in the comments.
I'll simplify by getting rid of this method.